File: java/src/main/java/S028.java
public class S028 {

    public static int solve() {
        return 1 + diagSum(3, 2, 500)
                + diagSum(5, 4, 500)
                + diagSum(7, 6, 500)
                + diagSum(9, 8, 500);
    }

    private static int diagSum(int start, int skip, int length) {
        int val = start;
        int delta = skip;
        int sum = val;
        for (int i = 1; i < length; i++) {
            delta += 8;
            val += delta;
            sum += val;
        }
        return sum;
    }

}

File: cpp/S028.cpp
#include <iostream>

using namespace std;

int diag_sum(int half_width) {
    int sum = 1;
    for (int i = 0; i < half_width; i++) {
        int to_sq = (i + 1) * 2 + 1;
        int sq = to_sq * to_sq;
        sum += (sq + sq - 3 * (to_sq - 1)) * 2;
    }
    return sum;
}

int main() {
    cout << diag_sum(500) << endl;
    return 0;
}
